src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1697,119 +2213,121 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
+                    }
 
                     List<Metric> metricList = metricProfile.getSloInfo().getFunctionVariables();
 
                     // Iterate over metrics and aggregation functions
                     for (Metric metricEntry : metricList) {
-                        HashMap<String, AggregationFunctions> aggregationFunctions = metricEntry.getAggregationFunctionsMap();
-                        for (Map.Entry<String, AggregationFunctions> aggregationFunctionsEntry: aggregationFunctions.entrySet()) {
-                            // Determine promQL query on metric type
-                            String metricQuery = aggregationFunctionsEntry.getValue().getQuery();
-                            String promQL = metricQuery;
-                            String format = null;
-
-
-                            // Determine format based on metric type - Todo move this metric profile
-                            List<String> cpuFunction = Arrays.asList(AnalyzerConstants.MetricName.cpuUsage.toString(), AnalyzerConstants.MetricName.cpuThrottle.toString(), AnalyzerConstants.MetricName.cpuLimit.toString(), AnalyzerConstants.MetricName.cpuRequest.toString());
-                            List<String> memFunction = Arrays.asList(AnalyzerConstants.MetricName.memoryLimit.toString(), AnalyzerConstants.MetricName.memoryRequest.toString(), AnalyzerConstants.MetricName.memoryRSS.toString(), AnalyzerConstants.MetricName.memoryUsage.toString());
-                            if (cpuFunction.contains(metricEntry.getName())) {
-                                format = KruizeConstants.JSONKeys.CORES;
-                            } else if (memFunction.contains(metricEntry.getName())) {
-                                format = KruizeConstants.JSONKeys.BYTES;
-                            }
-
-                            promQL = promQL
-                                    .replace(AnalyzerConstants.NAMESPACE_VARIABLE, namespace)
-                                    .replace(AnalyzerConstants.CONTAINER_VARIABLE, containerName)
-                                    .replace(AnalyzerConstants.MEASUREMENT_DURATION_IN_MIN_VARAIBLE, Integer.toString(measurementDurationMinutesInDouble.intValue()))
-                                    .replace(AnalyzerConstants.WORKLOAD_VARIABLE, workload)
-                                    .replace(AnalyzerConstants.WORKLOAD_TYPE_VARIABLE, workload_type);
+                        if (metricEntry.getName().startsWith(AnalyzerConstants.NAMESPACE) && !metricEntry.getName().equals("namespaceMaxDate")) {
+                            HashMap<String, AggregationFunctions> aggregationFunctions = metricEntry.getAggregationFunctionsMap();
+                            for (Map.Entry<String, AggregationFunctions> aggregationFunctionsEntry : aggregationFunctions.entrySet()) {
+                                String metricQuery = aggregationFunctionsEntry.getValue().getQuery();
+                                String promQL = metricQuery;
```

Comment:
update the variable assignation here as well

Reply:
updated here also.